src/operator/controllers/intents_reconcilers/network_policy.go

```diff
@@ -383,8 +383,11 @@ func (r *NetworkPolicyReconciler) CleanPoliciesFromUnprotectedServices(ctx conte
 
 	protectedServersByNamespace := sets.Set[string]{}
 	for _, protectedService := range protectedServicesResources.Items {
-		serverName := otterizev1alpha2.GetFormattedOtterizeIdentity(protectedService.Spec.Name, namespace)
-		protectedServersByNamespace.Insert(serverName)
+		// skip protected services that are in deletion process
+		if protectedService.DeletionTimestamp == nil {
```

Comment:
This checks that they are not being deleted.

Instead, make it like an err check, negate the check. If they are being deleted (!= nil), then `continue`